dartagnan/src/main/java/com/dat3m/dartagnan/wmm/analysis/NativeRelationAnalysis.java

```diff
@@ -389,7 +393,9 @@ public Map<Relation, ExtendedDelta> visitAcyclicity(Acyclicity axiom) {
             } while (!current.isEmpty());
             newDisabled.retainAll(knowledge.getMaySet());
             logger.debug("disabled {} edges in {}ms", newDisabled.size(), System.currentTimeMillis() - t0);
-            return Map.of(rel, new ExtendedDelta(newDisabled, new MapEventGraph()));
+            return axiom.isNegated() || axiom.isFlagged() ?
+                Map.of() :
+                Map.of(rel, new ExtendedDelta(newDisabled, new MapEventGraph()));
```

Comment:
You should put the check at the beginning and do an early return (also for the other two methods).
This function here computes a whole lot of stuff and even logs output if hte axiom is false/negated, all of which is unnecessary/wrong.